### PR TITLE
refactor(payment): INT-6964 Sunset Amazon Pay V1

### PR DESCRIPTION
**Pull request by mauricio-sg**
## What? [INT-6964](https://bigcommercecloud.atlassian.net/browse/INT-6964)
Remove all of the code related to Amazon Pay V1.

## Why?
So that we don’t have to continue maintaining.

## Testing / Proof
CircleCI

## Depends on
👉 https://github.com/bigcommerce/checkout-js/pull/1130

@bigcommerce/apex-integrations @bigcommerce/checkout @bigcommerce/payments

[INT-6964]: https://bigcommercecloud.atlassian.net/browse/INT-6964?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by josesonora — APPROVED** [review]

**Comment by bc-lessgarmi** [comment]
:green_heart: Tested!
**QE** details:
The task for AmazonPay has been validated without issue.

The following was tested:
• Regression Amazon Pay

Please see the attachment word below with the evidence:
https://drive.google.com/drive/folders/1dilqgH7eA5eFwMljWf9lS39aeR04CH4W?usp=share_link 

Please let me know any comment or doubt

**Review by Tharaae — APPROVED** [review]
LGTM

File: packages/core/src/payment/payment-strategy-type.ts [hunk 1]
```diff
@@ -5,7 +5,6 @@ enum PaymentStrategyType {
     ADYENV3_GOOGLEPAY = 'googlepayadyenv3',
     AFFIRM = 'affirm',
     AFTERPAY = 'afterpay',
-    AMAZON = 'amazon',
     AUTHORIZENET_GOOGLE_PAY = 'googlepayauthorizenet',
     AMAZONPAYV2 = 'amazonpay',
```
Comment (animesh1987) [inline packages/core/src/payment/payment-strategy-type.ts:9]: Should we rename this to `AMAZONPAY` now?

Comment (mauricio-sg) [reply packages/core/src/payment/payment-strategy-type.ts]: Done.

**Comment by mauricio-sg** [comment]
@animesh1987, your comments were addressed ♻️

**Review by animesh1987 — APPROVED** [review]

**Pull request merged**